### PR TITLE
add docs for devolping a slack dev bot

### PR DESCRIPTION
**Pull request by uncojohnco**
Upstream PR: #90 

- Add docs for creating a Slack dev bot

### CODE REVIEW

**Comment by uncojohnco** [comment]
@alysivji - do you have a preference for where assets for documentation should live?

**Comment by alysivji** [comment]
I think we should create a `docs` folder with an `images` or `assets` subfolder.

I envision the `docs` folder growing as we add features

**Comment by alysivji** [comment]
@uncojohnco looking great so far! ping me when you want me to review

File: doc/development-create-slack-bot.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+   <img src="images/install-app dev-bot 4 install-app-to-workspace-1.png" width=400>
+   </a></br>
+
+   Confirm the Authorization to Install the App to the Workspace. </br>
+   <a href="images/install-app dev-bot 4 install-app-to-workspace-2.png">
+   <img src="images/install-app dev-bot 4 install-app-to-workspace-2.png" width=400/>
+   </a>
+
+The `OAuth Access Token` should now be available. </br>
+These `OAuth Access Tokens` are used later on <TODO: link to step> </br>
```
Comment (uncojohnco) [inline doc/development-create-slack-bot.md]: add link to contribution.md

File: doc/development-create-slack-bot.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+<img src="images/install-app dev-bot 5 oauth-tokens.png" width=400/>
+
+### Enable Event Subscription
+
+- callback URL: generate via `ngrok`, see below
```
[the remark this answers is not in the document]
Comment (uncojohnco) [reply doc/development-create-slack-bot.md]: add link to contribution.md

File: doc/development-create-slack-bot.md [hunk 3]
```diff
@@ -0,0 +1,68 @@
+
+# Create a Slack Bot for Development
+
+## Create a Slack App
```
Comment (uncojohnco) [inline doc/development-create-slack-bot.md]: ## Init a Slack App

Comment (uncojohnco) [inline doc/development-create-slack-bot.md]: `## Init a Slack App`

File: doc/development-create-slack-bot.md [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+2. Define a name for the `Slack App`. </br>
+  <a href="images/create-a-slack-app 2.png">
+  <img src="images/create-a-slack-app 2.png" width=400/> 
+  </a></br>
+  After creating the `Slack App` will bring you to the `Basic Information` page</br>
+  <a href="images/create-a-slack-app 3.png">
+  <img src="images/create-a-slack-app 3.png" width=400/>
+  </a>
+
+### Install Slack App for bot development
```
Comment (uncojohnco) [inline doc/development-create-slack-bot.md]: `## Install Slack App`

File: doc/development-create-slack-bot.md [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+
+1. Install the dev-bot `Slack App` to Workspace. </br>
+  <a href="images/install-app dev-bot 1-1.png">
+  <img src="images/install-app dev-bot 1-1.png" width=400/>
+  </a>
+
+2. Add a `Bot User`. </br>
```
Comment (uncojohnco) [inline doc/development-create-slack-bot.md]: `## Add a Bot User`

File: doc/development-create-slack-bot.md [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+  <a href="images/install-app dev-bot 2 add-bot-user-1.png">
+  <img src="images/install-app dev-bot 2 add-bot-user-1.png"/ width=400> 
+  </a></br>
+  <a href="images/install-app dev-bot 2 add-bot-user-2.png">
+  <img src="images/install-app dev-bot 2 add-bot-user-2.png" width=400/>
+  </a>
+
+3. Define the `Slack App` `OAuth & Permissions` Scopes.
```
Comment (uncojohnco) [inline doc/development-create-slack-bot.md]: `## Define App OAuth & Permission`

File: doc/development-create-slack-bot.md [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+
+   Permission Scopes:
```
Comment (uncojohnco) [inline doc/development-create-slack-bot.md]: `1. Define the Permission Scopes for the App for the Dev-Bot`

File: doc/development-create-slack-bot.md [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+   - chat:write:bot
+   - bot
+   - usergroups:read
+   ```
+
+   <a href="images/install-app dev-bot 3 permission-scopes.png">
+   <img src="images/install-app dev-bot 3 permission-scopes.png" width=400/>
+   </a>
+
+4. Install `Slack App` to `Workspace` </br>
```
Comment (uncojohnco) [inline doc/development-create-slack-bot.md]: `##  Install App to Workspace`

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -29,30 +29,19 @@ It is recommended that users create a personal Slack workspace to use for bot de
...
 1. Install [Git-LFS](https://git-lfs.github.com/)
-1. Clone repo
-1. `cp .env.template .env`
-1. Populate `SLACK_BOTUSER_OAUTH_TOKEN` field with value obtained in previous section
-1. `make up`
-1. `make ngrok` to publish the development server on the internet
-1. Populate `NGROK_BASE_URI` field with value from previous step
-1. `make up` to refresh environment variables inside of Busy-Beaver
-1. Update callback URL to `http://[random_hash].ngrok.io/slack-event-subscription` in Slack application settings
+2. Clone repo
```
[the remark this answers is not in the document]
Comment (alysivji) [reply CONTRIBUTING.md]: if you leave all the numbers 1, they auto-increment and you don't have to keep renumbering if you add things in the middle.

Think if you have a space (like you do below with the steps with pictures), that trick doesn't work

[the remark this answers is not in the document]
Comment (uncojohnco) [reply CONTRIBUTING.md]: it looks live vscode/or extension has a lint to auto increment an ordered list =S

**Comment by uncojohnco** [comment]
hmm not just yet, I've moved the images into their own folder and need to update the links :frowning:

**Comment by alysivji** [comment]
Do you mind renaming the folder to `docs` ?

**Comment by uncojohnco** [comment]
@alysivji - the content for these docs are good to review. I've vetted the steps up till verifying the installation of the dev environment and looks to work following the instructions.

There are some minor styling issues as the markdown in vscode and github aren't on parity :frowning:

**Pull request merged**